pkg/logql/syntax/ast.go

```diff
@@ -54,6 +54,19 @@ func MustClone[T Expr](e T) T {
 	return copied
 }
 
+func ExtractLineFilterExprs[T Expr](e T) []LineFilterExpr {
```

Comment:
Not sure if this is the right place for this function, since it is the concrete usage of the `DepthFirstTraversal` struct.

Reply:
Also, are generics really needed here?

Reply:
I was following the same approach as other functions here. Example: `MustClone[T Expr](e T) T`